apm-sniffer/apm-agent/src/main/java/org/apache/skywalking/apm/agent/SkyWalkingAgent.java

```diff
@@ -130,6 +130,7 @@ public static void premain(String agentArgs, Instrumentation instrumentation) th
                     .installOn(instrumentation);
 
         try {
+            ServiceManager.INSTRUMENTATION = instrumentation;
```

Comment:
Why this?

Reply:
In order to make it more convenient to use instrumentation where necessary.

Reply:
This should not expose to only other developers. If anyone wants, build their own agent, it is not SkyWalking responsibility.